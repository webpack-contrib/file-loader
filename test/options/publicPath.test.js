/* eslint-disable
  prefer-destructuring,
*/
import webpack from '../helpers/compiler';

describe('Options', () => {
  describe('publicPath', () => {
    test('{String}', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            publicPath: 'public_path/',
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { assets, source } = stats.toJson().modules[1];

      expect({ assets, source }).toMatchSnapshot();
    });

    test('{String} - Without trailing slash', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            publicPath: 'public_path',
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { assets, source } = stats.toJson().modules[1];

      expect({ assets, source }).toMatchSnapshot();
    });

    test('{String} - URL', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            publicPath: 'https://cdn.com/',
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { assets, source } = stats.toJson().modules[1];

      expect({ assets, source }).toMatchSnapshot();
    });

    test('{Function}', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            publicPath(url) {
              return `public_path/${url}`;
            },
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { assets, source } = stats.toJson().modules[1];

      expect({ assets, source }).toMatchSnapshot();
    });
  });
});
