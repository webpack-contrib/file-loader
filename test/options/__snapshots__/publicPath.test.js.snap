// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options publicPath {Function} 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = \\"public_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`Options publicPath {String} - URL 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = \\"https://cdn.com/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`Options publicPath {String} - Without trailing slash 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = \\"public_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`Options publicPath {String} 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = \\"public_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;
